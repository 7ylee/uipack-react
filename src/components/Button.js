import React from 'react';
import PropTypes from 'prop-types';

export const style = {
    default: {
        'background': 'purple',
        'color': '#fff',
        'border': '1px solid #999',
        'padding': '0.25em 1em',
        'lineHeight': '1.5',
    },
    submit: {
        'background': '#fff',
        'color': '#666',
        'border': '1px solid #666',
        'padding': '0.25em 1em',
        'lineHeight': '1.5',
    }
};

/* eslint-disable react/button-has-type */
const Button = ({ type, children }) => (
    <button type={type} style={type === 'button' ? style.default : style.submit}>{children}</button>
);

// has default type
Button.defaultProps = {
    type: 'button'
};

Button.propTypes = {
    type: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired
};
export default Button;