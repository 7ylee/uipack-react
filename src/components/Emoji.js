import React from 'react';
import PropTypes from 'prop-types';

const Emoji = ({ label, content }) => (
    <span role="img" aria-label={label}>{content}</span>
);

Emoji.propTypes = {
    content: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired
};

export default Emoji;