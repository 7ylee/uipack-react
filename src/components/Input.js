import styled from 'styled-components';

const Input = styled.input`
    display: inline-block;
    font-size: 1em;
    padding: 0.25em 1em;
    border-radius: 3px;
    color: purple;
    border: 2px solid purple;
    box-shadow: none;

    &:focus {
        outline: none;
        box-shadow: 0 0 2px 2px #efefef
    }
`;

export default Input;