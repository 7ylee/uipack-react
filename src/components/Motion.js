import PropTypes from 'prop-types';
import React, { Component } from 'react';
import FromBottom from './Motion/FromBottom';

class Motion extends Component {
    static FromBottom = FromBottom;

    render() {
        const { isVisible, children } = this.props;

        return (
            <div>
                {React.Children.map(children, child =>
                    React.cloneElement(child, { animate: isVisible })
                )}
            </div>
        );
    }
}

Motion.defaultProps = {
    isVisible: undefined,
}

Motion.propTypes = {
    isVisible: PropTypes.bool,
    children: PropTypes.node.isRequired,
}

export default Motion;