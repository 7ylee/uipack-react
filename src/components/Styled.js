import styled from 'styled-components';

export const Btn = styled.button`
    background-color: #7D7D7D;
    color: #FFF;
`;

export const Segment = styled.div`
    background: #FFF;
`;
