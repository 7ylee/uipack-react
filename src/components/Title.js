import styled from 'styled-components';
import { resetCSS } from '../utility';

const Title = styled.h1`
    ${resetCSS};
    box-sizing: border-box;
    color: #333;
    display: block;
    font-size: ${props => props.sub ? '1.5em' : '2em'};
    border-bottom: ${props => props.underline && '1px solid #efefef'};
`;

export default Title;