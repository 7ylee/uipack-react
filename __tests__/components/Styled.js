import React from 'react';
import renderer from 'react-test-renderer';
import { Btn, Segment } from '../../src/components/Styled';

describe('Btn', () => {
    let tree;
    beforeEach(() => {
        tree = renderer
            .create(<Btn>AA</Btn>)
            .toJSON();
    })
    it('renders correctly', () => expect(tree).toMatchSnapshot());
});

describe('Segment', () => {
    let tree;
    beforeEach(() => {
        tree = renderer
            .create(<Segment>Segment</Segment>)
            .toJSON();
    })
    it('renders correctly', () => expect(tree).toMatchSnapshot());
});

