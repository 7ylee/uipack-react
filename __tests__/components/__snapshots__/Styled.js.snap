// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn renders correctly 1`] = `
<button
  className="sc-bdVaJa gApSIA"
>
  AA
</button>
`;

exports[`Segment renders correctly 1`] = `
<div
  className="sc-bwzfXH iCiuHk"
>
  Segment
</div>
`;
